src/builder.py

```diff
@@ -275,19 +389,86 @@ def run(config: state.BuilderRunConfig, proxy: state.ProxyConfig | None) -> list
         raise BuilderRunError from exc
 
 
-def get_latest_image(arch: state.Arch, base: state.BaseImage, cloud_name: str) -> str:
+def _get_image_name(arch: state.Arch, base: state.BaseImage, prefix: str) -> str:
+    """Get image name.
+
+    Args:
+        arch: The image compute architecture.
+        base: The Ubuntu OS base of the image.
+        prefix: The prefix to use for image name.
+
+    Returns:
+        The image name to upload on OpenStack.
+    """
+    return f"{prefix}-{base.value}-{arch.value}"
+
+
+def get_latest_images(config: state.BuilderRunConfig, cloud_id: str) -> list[CloudImage]:
```

Comment:
Does this method really needs to be using `multiprocessing` ? Is it heavily IO bound ?

Reply:
This is IO bound - not much CPU work here to be done.
I really wish we could use async await in charms but since we can't, it requires multiprocessing otherwise it will take hours for the images to be propagated unfortunately